py3dtiles/gltf_material.py

```diff
@@ -0,0 +1,71 @@
+class GlTFMaterial():
+    def __init__(self, metallicFactor=0, roughnessFactor=0, rgb=[1, 1, 1], alpha=1):
+        self.metallicFactor = metallicFactor
+        self.roughnessFactor = roughnessFactor
+        self.rgba = rgb
+        if len(rgb) < 4:
+            self.alpha = alpha
+
+    @property
+    def metallicFactor(self):
+        return self._metallicFactor
+
+    @metallicFactor.setter
+    def metallicFactor(self, value):
+        if value > 1:
+            value = 1
+        if value < 0:
+            value = 0
+        self._metallicFactor = value
+
+    @property
+    def roughnessFactor(self):
+        return self._roughnessFactor
+
+    @roughnessFactor.setter
+    def roughnessFactor(self, value):
+        if value > 1:
+            value = 1
+        if value < 0:
+            value = 0
+        self._roughnessFactor = value
+
+    @property
+    def rgba(self):
+        return self._rgba
+
+    @rgba.setter
+    def rgba(self, values):
+        for value in values:
+            if value > 1:
+                value /= 255
```

Comment:
Depending on the initial value of `value` the result of `/= 255` might not be normalized.
Maybe using the above mentioned `NormalizeData()` does not achieve the desired semantics (that I didn't try to grasp) but at least its always normalized...